cache/models/utils.js

```diff
@@ -39,34 +39,49 @@ function applyFieldListFilter(resource, attributes) {
  * @returns {Array} - Filtered resource objects array
  */
 function applyFiltersOnResourceObjs(resourceObjArray, opts) {
-  let filteredResources = resourceObjArray;
+  let filtered = resourceObjArray;
 
   // apply wildcard expr if other than name because
   // name filter was applied before redis call
   if (opts.filter) {
     const filterOptions = opts.filter;
     Object.keys(filterOptions).forEach((field) => {
-      if (field !== 'name') {
-        const filteredKeys = filterByFieldWildCardExpr(
-          resourceObjArray, field, filterOptions[field]
-        );
-        filteredResources = filteredKeys;
+      const value = filterOptions[field];
+      if (field === 'isPublished') {
+        filtered = filterByIsPublished(filtered, value);
+      } else if (field === 'tags') {
+        filtered = filterByTags(filtered, value);
+      } else if (field !== 'name' && typeof field === 'string') {
+        filtered = filterByFieldWildCardExpr(filtered, field, value);
       }
     });
   }
 
-  // sort and apply limits to resources
+  // sort resources
   if (opts.order) {
-    const sortedResources = sortByOrder(filteredResources, opts.order);
-    filteredResources = sortedResources;
+    filtered = sortByOrder(filtered, opts.order);
+  }
 
-    const slicedResourceObjs = applyLimitAndOffset(opts, filteredResources);
-    filteredResources = slicedResourceObjs;
+  // apply limits to resources if not already applied in prefilterKeys
+  if (!applyLimitAndOffsetInPrefilter(opts)) {
```

Comment:
All the functions that are been used here, were used before it was declared. While I do not have any preference, we have been declaring a function before using it.

Reply:
Previously, applyLimitAndOffset was being called before it was declared. Is there something in the style guide about this? 

I definitely prefer being able to declare functions in any order.

Reply:
Sorry, I did not notice your reply earlier. My linter complains about this every time. Looks like this is something to do with the `no-use-before-define` rule.

Reply:
This would be an outlier in our codebase (except for some views) -- let's please stick with defining functions before using them.